### PR TITLE
attrib.d: use foreachDsymbol()

### PR DESCRIPTION
**Pull request by WalterBright**
I love deleting boilerplate.

### CODE REVIEW

File: src/dmd/attrib.d [hunk 1]
```diff
@@ -54,20 +54,7 @@ extern (C++) abstract class AttribDeclaration : Dsymbol
...
-                Dsymbol s = (*d)[i];
-                if (s)
-                {
-                    if (s.apply(fp, param))
-                        return 1;
-                }
-            }
-        }
-        return 0;
+        return include(_scope).foreachDsymbol( (s) { return s && s.apply(fp, param); } );
```
Comment (ZombineDev) [inline src/dmd/attrib.d]: Use:

```suggestion
        return include(_scope).foreachDsymbol(s => s && s.apply(fp, param));
```

Or at least:

```suggestion
        return include(_scope).foreachDsymbol( s => s && s.apply(fp, param) );
```

If you demand on having whitespace around the lambda

File: src/dmd/attrib.d [hunk 2]
```diff
@@ -193,73 +158,29 @@ extern (C++) abstract class AttribDeclaration : Dsymbol
...
-        {
-            for (size_t i = 0; i < d.dim; i++)
-            {
-                Dsymbol s = (*d)[i];
-                if (s.hasPointers())
-                    return true;
-            }
-        }
-        return false;
+        return include(null).foreachDsymbol( (s) { return s.hasPointers(); } ) != 0;
```
Comment (ZombineDev) [inline src/dmd/attrib.d]:
```suggestion
        return !!include(null).foreachDsymbol(s => s.hasPointers);
```

or: 

```suggestion
        return include(null).foreachDsymbol(s => s.hasPointers) != 0;
```

**Review by PetarKirov — APPROVED** [review]
Yeah, looks pretty good indeed.

File: src/dmd/attrib.d (hunk 1 above)
Comment (WalterBright) [reply src/dmd/attrib.d]: Again, the `return` is needed for the early out loop.

File: src/dmd/attrib.d (hunk 2 above)
Comment (WalterBright) [reply src/dmd/attrib.d]: ditto

File: src/dmd/attrib.d (hunk 1 above)
Comment (ZombineDev) [reply src/dmd/attrib.d]: The two syntaxes have equivalent semantics, as far as I know. You would still break from the loop as soon as `apply` retuens non-zero.

**Review by thewilsonator — APPROVED** [review]

**Pull request merged**